### DISCUSSION TITLE
How do I use argon2?

### DISCUSSION THREAD

**Discussion by Chippel** [dyne/tomb#517] [General]
Hey all, afaik argon2 is the preffered algorithm compared to pbkd. I'd like to enable it but I don't know if the following is enough

--kdftype 

is this enough or do I have to add argon2 afterwards?

**Comment by Narrat** [comment]
The content of the option is checked, so add `argon2` afterwards. If `--kdftype` is empty it should fall back to `pbkdf`

**Comment by Chippel** [comment]
Why is it not enabled by default? It should be much more secure shouldn’t it?

**Reply by Narrat** [reply]
Theoretically `argon2` covers against cases, where `kdf` is lacking, yes.
But as far as I know kdf isn't discouraged to use (or broken). Also there may be distributions or os where argon2 isn't available yet, which would make it a bad choice as a default. Memory constraint systems could be another argument. And backwards compatible behaviour another.
